flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/casting/CastRulesTest.java

```diff
@@ -350,34 +356,63 @@ private static CastTestSpecBuilder testCastTo(DataType targetType) {
         }
 
         private CastTestSpecBuilder fromCase(DataType dataType, Object src, Object target) {
-            this.inputTypes.add(dataType);
-            this.inputData.add(src);
-            this.expectedValues.add(target);
-            this.castContexts.add(
+            return fromCase(
+                    dataType,
                     CastRule.Context.create(
-                            ZoneId.systemDefault(),
```

Comment:
Let's use UTC here instead. Relying on system defaults makes tests non-deterministic in different CI systems.